### PR TITLE
add prometheus crds and fix kit permission

### PR DESCRIPTION
**Pull request by bwagner5**
Issue #, if available:
N/A

Description of changes:
 - Add prometheus podmonitor CRD 
 - Fix security group permission for KIT
 - Add test namespace 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: infrastructure/k8s-config/clusters/kit-infrastructure/kit-provisioner.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  subnetSelector:
+    kit.sh/stack: KITInfrastructure
+    aws-cdk:subnet-type: Private
+  securityGroupSelector:
+    kit.sh/stack: KITInfrastructure
+---
+apiVersion: karpenter.sh/v1alpha5
+kind: Provisioner
+metadata: 
+  name: kit
```
Comment (prateekgogia) [inline infrastructure/k8s-config/clusters/kit-infrastructure/kit-provisioner.yaml:16]: Why do we need another provisioner? Can't we extend the existing [default provisioner](https://github.com/awslabs/kubernetes-iteration-toolkit/blob/main/infrastructure/k8s-config/clusters/kit-infrastructure/provisioner.yaml)?

Having a KIT provisioner, does it mean we will need to update the pods for guest clusters which rely today on default provisioner to provision nodes?

Comment (prateekgogia) [reply infrastructure/k8s-config/clusters/kit-infrastructure/kit-provisioner.yaml:16]: Had an offline discussion, we don't need to update guest cluster pods to provision nodes using KIT provisioner with the latest Karpenter version.

**Review by prateekgogia — APPROVED** [review]
LGTM!
Next task will be to upgrade karpenter installed as part of kitctl. Also use taints and tolerations for KIT pods to only be scheduled on dedicated KIT nodes.

File: infrastructure/k8s-config/clusters/kit-infrastructure/kit-provisioner.yaml (hunk 1 above)
Comment (hakuna-matatah) [reply infrastructure/k8s-config/clusters/kit-infrastructure/kit-provisioner.yaml:16]: Then what is the purpose of this provisioner being here ?

**Pull request merged**